### PR TITLE
Story 619

### PR DESCRIPTION
**Pull request by deepakr123**

### CODE REVIEW

File: controllers/v1/assessmentsController.js [hunk 1]
```diff
@@ -277,6 +277,7 @@ module.exports = class Assessments {
                     frameworkExternalId: solutionDocument.frameworkExternalId,
                     entityTypeId: solutionDocument.entityTypeId,
                     entityType: solutionDocument.entityType,
+                    scoringSystem : solutionDocument.scoringSystem,
```
Comment (aks30) [inline controllers/v1/assessmentsController.js:280]: @deepakr123 - Add isRubricDriven key in the submission document.

File: generics/messageConstants/apiResponses.js [hunk 2]
```diff
@@ -204,5 +204,9 @@ module.exports = {
 "LIBRARY_CATEGORIES_NOT_FOUND" : "Could not found library categories",
 "URL_COULD_NOT_BE_FOUND" : "Could not found url from gcp",
 "KENDRA_SERVICE_DOWN" : "Kendra service is down",
-"SUBMISSION_LIST_FETCHED" : "Submission list fetched successfully"
+"SUBMISSION_LIST_FETCHED" : "Submission list fetched successfully",
+"CRITERIA_QUESTIONS_FETCHED_SUCCESSFULLY" : "Criteria questions fetched successfully",
+"SUCCESSFULLY_SUBMITTED_MANUAL_RATING" : "submitted the manual rating successfully",
```
Comment (aks30) [inline generics/messageConstants/apiResponses.js:209]: @deepakr123 - Rename it to to - "Manual rating submitted successfully."

File: generics/messageConstants/apiResponses.js [hunk 3]
```diff
@@ -204,5 +204,9 @@ module.exports = {
...
+"RATING_SUBMISSION_FAILURE" : "Could not submit the rating",
```
Comment (aks30) [inline generics/messageConstants/apiResponses.js:210]: @deepakr123 - Change it to "Rating could not be submitted."

File: generics/messageConstants/apiResponses.js [hunk 4]
```diff
@@ -204,5 +204,9 @@ module.exports = {
...
+"FETCH_CRITERIA_QUESTIONS_FAILURE" : "rating is already done / scoring system has to be manual"
```
Comment (aks30) [inline generics/messageConstants/apiResponses.js:211]: @deepakr123 - Change it to - "Rating is already submitted or rating submission is not allowed."

File: generics/messageConstants/common.js [hunk 5]
```diff
@@ -18,5 +18,9 @@ module.exports = {
     PUBLISHED : "published",
     ACTIVE_STATUS : "active",
     STARTED : "started",
-    LEAD_ASSESSOR : "LEAD_ASSESSOR"
+    LEAD_ASSESSOR : "LEAD_ASSESSOR",
+    MANUAL_RATING : "manual",
+    AUTO_RATING : "auto",
+    SUBMISSION_STATUS : "ratingPending",
```
Comment (aks30) [inline generics/messageConstants/common.js:24]: @deepakr123 - Add constants for all submission status

File: models/submissions.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ module.exports = {
       type : Boolean
     },
     submissionNumber : Number,
+    scoringSystem : String,
```
Comment (aks30) [inline models/submissions.js:39]: @deepakr123 - Add isRubricDriven key as well and in observation submissions

File: module/criteria/helper.js [hunk 7]
```diff
@@ -57,7 +57,7 @@ module.exports = class criteriaHelper {
                     {_id : criteriaId},
                     {
                         rubric: rubric,
-                        criteriaType : "auto"
+                        criteriaType : existingCriteria.criteriaType ? existingCriteria.criteriaType : messageConstants.common.AUTO_RATING
```
Comment (aks30) [inline module/criteria/helper.js:60]: @deepakr123 - This will always be auto

Comment (aks30) [reply module/criteria/helper.js:60]: @deepakr123 - Add a condition to check only allowed levels are submitted in CSV.

File: module/submissions/validator/v1.js [hunk 8]
```diff
@@ -25,7 +25,13 @@ module.exports = (req) => {
             .isMongoId().withMessage("Invalid solution id");
             req.checkQuery('entityId').exists().withMessage("required entity id")
             .isMongoId().withMessage("Invalid entity id");
-        }   
+        },
+        getCriteriaQuestions: function () {
+            req.checkParams('_id').exists().withMessage("required submission id");
```
Comment (aks30) [inline module/submissions/validator/v1.js:30]: @deepakr123 - Add validation of whether or not it is a valid mongoID or not.

File: module/submissions/validator/v1.js [hunk 9]
```diff
@@ -25,7 +25,13 @@ module.exports = (req) => {
...
+        },
+        manualRating: function () {
+            req.checkParams('_id').exists().withMessage("required submission id");
```
Comment (aks30) [inline module/submissions/validator/v1.js:33]: @deepakr123 - Validation if post body is there and that at least one criteria in the post body

File: module/submissions/helper.js [hunk 10]
```diff
@@ -1483,4 +1483,326 @@ module.exports = class SubmissionsHelper {
...
 
+    /**
+    * Get criteria questions 
+    * @method
+    * @name getCriteriaQuestions
+    * @param {String} submissionId - submissionId.
+    * @returns {Object} - Criteria questions and answers 
+    */
+
+    static getCriteriaQuestions(submissionId) {
```
Comment (aks30) [inline module/submissions/helper.js:1494]: @deepakr123 - Check for if submissionId is not passed condition.

File: controllers/v1/assessmentsController.js (hunk 1 above)
Comment (deepakr123) [reply controllers/v1/assessmentsController.js:280]: resolved

File: generics/messageConstants/apiResponses.js (hunk 4 above)
Comment (deepakr123) [reply generics/messageConstants/apiResponses.js:211]: resolved

File: generics/messageConstants/apiResponses.js (hunk 3 above)
Comment (deepakr123) [reply generics/messageConstants/apiResponses.js:210]: resolved

File: generics/messageConstants/apiResponses.js (hunk 2 above)
Comment (deepakr123) [reply generics/messageConstants/apiResponses.js:209]: resolved

File: generics/messageConstants/common.js (hunk 5 above)
Comment (deepakr123) [reply generics/messageConstants/common.js:24]: resolved

File: models/submissions.js (hunk 6 above)
Comment (deepakr123) [reply models/submissions.js:39]: resolved

File: module/submissions/validator/v1.js (hunk 8 above)
Comment (deepakr123) [reply module/submissions/validator/v1.js:30]: resolved

File: module/submissions/validator/v1.js (hunk 9 above)
Comment (deepakr123) [reply module/submissions/validator/v1.js:33]: resolved

File: module/submissions/helper.js (hunk 10 above)
Comment (deepakr123) [reply module/submissions/helper.js:1494]: resolved

File: controllers/v1/submissionsController.js [hunk 11]
```diff
@@ -2561,4 +2561,127 @@ module.exports = class Submission extends Abstract {
...
+        });
+      }
+    })
+  }
+
+
+  /**
+  * @api {post} /assessment/api/v1/submissions/manualRating/:submissionId
+  * @apiVersion 1.0.0
+  * @apiName manual rating
```
Comment (aks30) [inline controllers/v1/submissionsController.js:2642]: @deepakr123 - API name has to start with a capital

File: controllers/v1/submissionsController.js [hunk 12]
```diff
@@ -2561,4 +2561,127 @@ module.exports = class Submission extends Abstract {
...
+  * {
+  *  "5698fa069f664f7e1ae7499d" : "L1",
+  *  "58673e7b9f664f7e1ae7388e" : "L2"
+  * }
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Response:
+  * {
+     "status": 200,
+     "message": "submitted the manual rating successfully"
```
Comment (aks30) [inline controllers/v1/submissionsController.js:2655]: @deepakr123 - Change the message here please.

File: controllers/v1/submissionsController.js [hunk 13]
```diff
@@ -2561,4 +2561,127 @@ module.exports = class Submission extends Abstract {
...
+   * @param {String} req.params._id - submission id. 
+   * @returns {JSON} manual rating
+   */
+
+  async manualRating(req) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        
+        let response =
+        await submissionsHelper.manualRating( req );
```
Comment (aks30) [inline controllers/v1/submissionsController.js:2673]: @deepakr123 - Pass 2 parameters, submission id and the body

File: generics/messageConstants/common.js [hunk 14]
```diff
@@ -18,5 +18,11 @@ module.exports = {
...
     ACTIVE_STATUS : "active",
     STARTED : "started",
-    LEAD_ASSESSOR : "LEAD_ASSESSOR"
+    LEAD_ASSESSOR : "LEAD_ASSESSOR",
+    MANUAL_RATING : "manual",
+    AUTO_RATING : "auto",
+    SUBMISSION_STATUS_RATING_PENDING: "ratingPending",
+    SUBMISSION_STATUS_COMPLETED : "completed",
+    SUBMISSION_STATUS_STARTED : "started",
+    SUBMISSION_STATUS_IN_PROGRESS : "inProgess"
```
Comment (aks30) [inline generics/messageConstants/common.js:27]: @deepakr123 - It will be "inprogress"

File: routes/index.js [hunk 15]
```diff
@@ -89,7 +89,8 @@ module.exports = function (app) {
             totalCount: result.totalCount,
             total: result.total,
             count: result.count,
-            failed: result.failed
+            failed: result.failed,
+            success: result.success
```
Comment (aks30) [inline routes/index.js:93]: @deepakr123 - This is not required.

File: controllers/v1/submissionsController.js (hunk 11 above)
Comment (deepakr123) [reply controllers/v1/submissionsController.js:2642]: resolved

File: controllers/v1/submissionsController.js (hunk 12 above)
Comment (deepakr123) [reply controllers/v1/submissionsController.js:2655]: resolved

File: controllers/v1/submissionsController.js (hunk 13 above)
Comment (deepakr123) [reply controllers/v1/submissionsController.js:2673]: resolved

File: generics/messageConstants/common.js (hunk 14 above)
Comment (deepakr123) [reply generics/messageConstants/common.js:27]: resolved

File: routes/index.js (hunk 15 above)
Comment (deepakr123) [reply routes/index.js:93]: resolved

**Pull request merged**